routing/payment_lifecycle.go

```diff
@@ -190,7 +190,11 @@ func (p *paymentLifecycle) resumePayment() ([32]byte, *route.Route, error) {
 // payment-level failure.
 func errorToPaymentFailure(err error) channeldb.FailureReason {
 	switch err {
-	case errNoTlvPayload, errNoPathFound, errMaxHopsExceeded,
+	case
+		errNoTlvPayload,
+		errNoPaymentAddr,
+		errNoPathFound,
+		errMaxHopsExceeded,
 		errPrebuiltRouteTried:
 
 		return channeldb.FailureReasonNoRoute
```

Comment:
Maybe later we want to reconsider the mapping of all of this to `NoRoute`

Reply:
NoTlvPayload for the final node isn't really a `NoRoute` result. `InvalidDetails` would cover the situation better probably. But doesn't need to be discussed/changed now.